File: tier1/python/83e733f6/handler.py
import os


BASE = "/srv/reports"


def handle(name):
    target = os.path.realpath(os.path.join(BASE, os.path.basename(name)))
    if not target.startswith(BASE + os.sep):
        return ""
    with open(target) as handle_:
        return handle_.read()

File: tier1/javascript/db584049/handler.js
const fs = require('fs');
const path = require('path');

const BASE = '/srv/reports';

function handle(name) {
  const target = path.resolve(BASE, path.basename(name));
  if (!target.startsWith(BASE + path.sep)) {
    return '';
  }
  return fs.readFileSync(target, 'utf8');
}

module.exports = { handle };
